hamilton/experimental/h_databackends.py

```diff
@@ -0,0 +1,161 @@
+
+# TODO add a `_has__dataframe__` attribute for those that implement the
+# dataframe interchange protocol
+
+
+# PyArrow
+class AbstractPyArrowDataFrame(AbstractBackend):
+    _backends = [("pyarrow", "Table"), ("pyarrow", "RecordBatch")]
+
+
+class AbstractPyarrowColumn(AbstractBackend):
+    _backends = [("pyarrow", "Array"), ("pyarrow", "ChunkedArray")]
+
+
+# Ibis
+class AbstractIbisDataFrame(AbstractBackend):
+    _backends = [("ibis.expr.types", "Table")]
+
+
+class AbstractIbisColumn(AbstractBackend):
+    _backends = [("ibis.expr.types", "Column")]
+
+
+# Pandas
+class AbstractPandasDataFrame(AbstractBackend):
+    _backends = [("pandas", "DataFrame")]
+
+
+class AbstractPandasColumn(AbstractBackend):
+    _backends = [("pandas", "Series")]
+
+
+# Polars
+class AbstractPolarsDataFrame(AbstractBackend):
+    _backends = [("polars", "DataFrame")]
+
+
+class AbstractPolarsColumn(AbstractBackend):
+    _backends = [("polars", "Series")]
+
+
+# Polars Lazy
+class AbstractLazyPolarsDataFrame(AbstractBackend):
+    _backends = [("polars", "LazyFrame")]
+
+
+# Vaex
+class AbstractVaexDataFrame(AbstractBackend):
+    _backends = [("vaex.dataframe", "DataFrame")]
+
+
+class AbstractVaexColumn(AbstractBackend):
+    _backends = [("vaex.expression", "Expression")]
+
+
+# Dask
+class AbstractDaskDataFrame(AbstractBackend):
+    _backends = [("dask.dataframe", "DataFrame")]
+
+
+class AbstractDaskColumn(AbstractBackend):
+    _backends = [("dask.dataframe", "Series")]
+
+
+# SparkSQL
+class AbstractSparkSQLDataFrame(AbstractBackend):
+    _backends = [("pyspark.sql", "DataFrame")]
+
+
+# SparkPandas
+class AbstractSparkPandasDataFrame(AbstractBackend):
+    _backends = [("pyspark.pandas", "DataFrame")]
+
+
+class AbstractSparkPandasColumn(AbstractBackend):
+    _backends = [("pyspark.pandas", "Series")]
+
+
+# Geopandas
+class AbstractGeoPandasDataFrame(AbstractBackend):
+    _backends = [("geopandas", "GeoDataFrame")]
+
+
+class AbstractGeoPandasColumn(AbstractBackend):
+    _backends = [("geopandas", "GeoSeries")]
+
+
+# cuDF
+class AbstractCuDFDataFrame(AbstractBackend):
+    _backends = [("cudf", "DataFrame")]
+
+
+# Modin
+class AbstractModinDataFrame(AbstractBackend):
+    _backends = [("modin.pandas", "DataFrame")]
+
+
+def register_backends() -> Tuple[Type, Type]:
+    """Register Abstract classes"""
+    abstract_dataframe_types = set()
+    abstract_column_types = set()
+
+    h_databackends_module = importlib.import_module(__name__)
+    for name, cls in inspect.getmembers(h_databackends_module, inspect.isclass):
+        if "DataFrame" in name:
+            abstract_dataframe_types.add(cls)
+        elif "Column" in name:
+            abstract_column_types.add(cls)
+
+    # overwrite sets with Union type objects
+    DATAFRAME_TYPES = Union[tuple(abstract_dataframe_types)]
+    COLUMN_TYPES = Union[tuple(abstract_column_types)]
+    return DATAFRAME_TYPES, COLUMN_TYPES
+
+
+DATAFRAME_TYPES, COLUMN_TYPES = register_backends()
```

Comment:
So, just to be clear, is this replacing our standard plugin methodology? Not 100% sure the value of this (although I did see your loom) -- need to look over it a bit closer.

Reply:
This could effectively replace the need to [dynamically register distributed plugin files](https://github.com/DAGWorks-Inc/hamilton/blob/770050751840318aa2ac923c0f3137c4412dfa2f/hamilton/registry.py#L76)